### PR TITLE
Improve access token response handling

### PR DESCRIPTION
**Pull request by wemrysi**
Fixes an issue when decoding the `scope` of a successful access token response and sanitizes any error messages to not contain the response text to avoid leaking credentials.

### CODE REVIEW

File: core/src/main/scala/quasar/datasource/url/oauth/types.scala [hunk 1]
```diff
@@ -85,21 +84,16 @@ object ClientSecret {
...
 
-final case class ScopeToken(value: String) extends AnyVal
-
-object ScopeToken {
-  implicit val codecJson: CodecJson[ScopeToken] =
-    CodecJson.derived[String].xmap(ScopeToken(_))(_.value)
-}
-
-final case class Scope(tokens: NonEmptyList[ScopeToken]) extends AnyVal
+final case class Scope(tokens: NonEmptyList[String]) extends AnyVal
```
Comment (rintcius) [inline core/src/main/scala/quasar/datasource/url/oauth/types.scala]: Ok so the no scope scenario mentioned here https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/#available-scopes would be handled by not supplying the scope field, right?

File: README.md [hunk 2]
```diff
@@ -92,7 +92,7 @@ Authentication via the OAuth 2.0 authorization code flow may be enabled by inclu
     "clientId": String,
     "clientSecret": String,
     "clientAuthenticationMethod": "basic" | "request-body"
-    [, "scope": [String]]
+    [, "scope": String]
```
Comment (rintcius) [inline README.md]: Ah so you changed the scope from array to string to match the RFC, right?

Comment (wemrysi) [reply README.md]: Yes, that seemed more straightforward.

File: core/src/main/scala/quasar/datasource/url/oauth/types.scala (hunk 1 above)
Comment (wemrysi) [reply core/src/main/scala/quasar/datasource/url/oauth/types.scala]: Correct :+1:

**Pull request merged**